src/components/CreateUserForm.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from "react";
+import { useMutation, useQuery } from "react-query";
+
+interface User {
+  firstname: string;
+  lastname: string;
+  email: string;
+  profession: string | null;
+}
+
+function CreateUserForm() {
+  const [firstname, setFirstname] = useState<string>("");
+  const [lastname, setLastName] = useState<string>("");
+  const [email, setEmail] = useState<string>("");
+  const [profession, setProfession] = useState<string>("");
```

Comment:
Replace `useState<string>("")` with `useState("")`

As you pass a string as default value, the type is infered
https://www.carlrippon.com/typed-usestate-with-typescript/

Reply:
at some point typescript complained about the type of the state being `never`

Reply:
It is the case if you don't give a default value iirc